### PR TITLE
Add configuratgion for Tls route

### PR DESCRIPTION
**Pull request by maphad**
Add configuration for traffic director tlsRoute resource

part of https://github.com/hashicorp/terraform-provider-google/issues/14040

If this PR is for Terraform, I acknowledge that I have:

- [ x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [ x] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [ x] [Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [`make test` and `make lint`](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.
- [ x] [Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/run-provider-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [ x] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:new-resource
`google_networkservices_tls_route`
```

### CODE REVIEW

**Comment by rileykarson** [comment]
Adding @NickElliot as initial reviewer (in a reverse shadow). Once they've approved, @hao-nan-li will review and give final approval.

File: mmv1/products/networkservices/TlsRoute.yaml
Comment (NickElliot) [inline mmv1/products/networkservices/TlsRoute.yaml:93]: Are these fields going to be used? create_time and update_time aren't normally needed fields in Terraform and could be removed if we don't expect a use case. update_time in particular is a transient value that could not be relied on in code.

File: mmv1/products/networkservices/TlsRoute.yaml
Comment (NickElliot) [inline mmv1/products/networkservices/TlsRoute.yaml:144]: Should this field be marked required too?

File: mmv1/products/networkservices/TlsRoute.yaml
Comment (maphad) [reply mmv1/products/networkservices/TlsRoute.yaml:93]: They have been marked as output only fields.

Comment (NickElliot) [reply mmv1/products/networkservices/TlsRoute.yaml:93]: Even as output only fields their inclusion is generally not needed or useful if there is no special reason to be including them

File: mmv1/products/networkservices/TlsRoute.yaml
Comment (NickElliot) [inline mmv1/products/networkservices/TlsRoute.yaml:105]: Could you please update this description to include the required format mentioned in the API reference?

File: mmv1/products/networkservices/TlsRoute.yaml
Comment (NickElliot) [inline mmv1/products/networkservices/TlsRoute.yaml:111]: Same "include required format" update for this one! Thanks

File: mmv1/products/networkservices/TlsRoute.yaml
Comment (NickElliot) [inline mmv1/products/networkservices/TlsRoute.yaml:131]: could the supported wildcard behavior be included in this description please?

File: mmv1/products/networkservices/TlsRoute.yaml
Comment (NickElliot) [inline mmv1/products/networkservices/TlsRoute.yaml:137]: It would also be good to update these to include the "At least one of sniHost and alpn is required." and "Up to 5" details

File: mmv1/products/networkservices/TlsRoute.yaml
Comment (NickElliot) [reply mmv1/products/networkservices/TlsRoute.yaml:93]: Even as output only fields, their inclusion is generally not useful or needed in the terraform state.

File: mmv1/products/networkservices/TlsRoute.yaml
Comment (maphad) [reply mmv1/products/networkservices/TlsRoute.yaml]: Added

File: mmv1/products/networkservices/TlsRoute.yaml
Comment (maphad) [reply mmv1/products/networkservices/TlsRoute.yaml]: Added

File: mmv1/products/networkservices/TlsRoute.yaml
Comment (maphad) [reply mmv1/products/networkservices/TlsRoute.yaml]: Added

File: mmv1/products/networkservices/TlsRoute.yaml
Comment (maphad) [reply mmv1/products/networkservices/TlsRoute.yaml]: Added

File: mmv1/products/networkservices/TlsRoute.yaml
Comment (maphad) [reply mmv1/products/networkservices/TlsRoute.yaml:93]: I would keep this for the sake of consistency with the majority of the resources.

**Comment by NickElliot** [comment]
LGTM! @hao-nan-li for final review.

File: mmv1/third_party/terraform/tests/resource_network_services_tls_route_test.go.erb
Comment (hao-nan-li) [inline mmv1/third_party/terraform/tests/resource_network_services_tls_route_test.go.erb:15]: Not really a big deal, it looks like there's some missed indentation on the test file. If you have time please align them, otherwise LGTM.

**Review by hao-nan-li — DISMISSED** [review]

**Review by hao-nan-li — COMMENTED** [review]
Looks like there are couple tests failing, some of them are flaky.
Could you make sure all the `TestAccNetworkServicesTlsRoute` work as expected?

**Comment by maphad** [comment]
> Looks like there are couple tests failing, some of them are flaky. Could you make sure all the `TestAccNetworkServicesTlsRoute` work as expected?

Looks like I had missed out on running tests locally after adding some new tests. I have fixed them now and verified via local run. Apologies for missing out on this!

**Review by hao-nan-li — APPROVED** [review]
LGTM.

**Pull request merged**
